File: icamera/src/main/java/me/shouheng/icamera/enums/CameraSizeFor.java
package me.shouheng.icamera.enums;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static me.shouheng.icamera.enums.CameraSizeFor.*;

/**
 * Camera size for preview, picture, video, etc.
 *
 * @author <a href="mailto:shouheng2015@gmail.com">WngShhng</a>
 * @version 2019-12-28 16:38
 */
@IntDef(value = {SIZE_FOR_PREVIEW, SIZE_FOR_PICTURE, SIZE_FOR_VIDEO})
@Retention(value = RetentionPolicy.SOURCE)
public @interface CameraSizeFor {

    /** Camera size for preview */
    int SIZE_FOR_PREVIEW    = 0x0010;

    /** Camera size for picture */
    int SIZE_FOR_PICTURE    = 0x0020;

    /** Camera size for video */
    int SIZE_FOR_VIDEO      = 0x0040;
}

File: icamera/src/main/java/me/shouheng/icamera/enums/CameraSizeFor.kt
package me.shouheng.icamera.enums

import androidx.annotation.IntDef

/**
 * Camera size for preview, picture, video, etc.
 *
 * @author [WngShhng](mailto:shouheng2015@gmail.com)
 * @version 2019-12-28 16:38
 */
@IntDef(value = [
    CameraSizeFor.SIZE_FOR_PREVIEW,
    CameraSizeFor.SIZE_FOR_PICTURE,
    CameraSizeFor.SIZE_FOR_VIDEO])
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class CameraSizeFor {
    companion object {
        /** Camera size for preview  */
        const val SIZE_FOR_PREVIEW = 0x0010

        /** Camera size for picture  */
        const val SIZE_FOR_PICTURE = 0x0020

        /** Camera size for video  */
        const val SIZE_FOR_VIDEO = 0x0040
    }
}